xmtp_mls/src/identity_updates.rs

```diff
@@ -75,6 +78,33 @@ where
 
         Ok(get_state(updates)?)
     }
+
+    pub fn get_association_state_diff<InboxId: AsRef<str>>(
+        &self,
+        conn: &'a DbConnection<'a>,
+        inbox_id: String,
+        starting_sequence_id: Option<i64>,
+        ending_sequence_id: Option<i64>,
+    ) -> Result<AssociationStateDiff, ClientError> {
+        let initial_state = self.get_association_state(conn, &inbox_id, starting_sequence_id)?;
+        if starting_sequence_id.is_none() {
+            return Ok(AssociationStateDiff::from(initial_state));
+        }
+
+        let incremental_updates = conn
+            .get_identity_updates(inbox_id, starting_sequence_id, ending_sequence_id)?
+            .into_iter()
+            .map(|update| update.try_into())
+            .collect::<Result<Vec<IdentityUpdate>, AssociationError>>()?;
+
+        let final_state = incremental_updates
+            .into_iter()
+            .try_fold(initial_state.clone(), |state, update| {
```

Comment:
Will this clone on every step of the iteration?

Reply:
It will. `apply_update` treats the existing state as immutable so that you can discard updates if they fail part-way.

I think in at least some cases (one way associations come to mind) we are going to want to move past invalid updates rather than failing the entire process.